### PR TITLE
Add PmtId->InstrId and Othr->SchmeNm

### PR DESCRIPTION
**Pull request by MTijn**

### CODE REVIEW

**Comment by MTijn** [comment]
This should solve https://github.com/php-sepa-xml/php-sepa-xml/issues/159

File: src/GroupHeader.php [hunk 1]
```diff
@@ -83,7 +83,7 @@ class GroupHeader
     /**
      * @var string
      */
-    protected $creationDateTimeFormat = 'Y-m-d\TH:i:s\Z';
+    protected $creationDateTimeFormat = \DateTime::RFC3339;
```
Comment (BorislavSabev) [inline src/GroupHeader.php:86]: This is not correct. The SEPA date format is not RFC3339. 
There are some notable differences - for e.g. RFC3339 allows that the `T` in the middle is replaced by a space.

**Review by BorislavSabev — CHANGES_REQUESTED** [review]

File: src/GroupHeader.php (hunk 1 above)
Comment (BorislavSabev) [reply src/GroupHeader.php:86]: Your example is only one case of many. 

It's just not the same standard and we cannot stray away from the ISO20022 standard. I am starting to think that the original standard as put in the library is also not an exact representation of the ISO20022 DateTime standard.

The main differences between the ISO 20022 DateTime format and RFC 3339 are:

1. Representation of Time Zone: ISO 20022 uses the format 'YYYY-MM-DDThh:mm:ss.sss±hh:mm' to represent the time zone, where the offset from UTC is included after the time. In contrast, RFC 3339 uses the format 'YYYY-MM-DDThh:mm:ss.sssZ' where the 'Z' at the end indicates the UTC time zone. [1](https://stackoverflow.com/questions/47283611/php-date-format-vs-symfony-rfc-3339-format)[2](https://ijmacd.github.io/rfc3339-iso8601/)

2. Precision of Fractional Seconds: ISO 20022 allows for up to three decimal places for fractional seconds, while RFC 3339 allows for up to nine decimal places. This provides higher precision in the RFC 3339 format. [2](https://ijmacd.github.io/rfc3339-iso8601/)

3. Readability: Some sources suggest that the RFC 3339 format is more readable and less ambiguous than the ISO 20022 format, as it uses the 'Z' character to clearly indicate the UTC time zone. [3](https://www.reddit.com/r/ISO8601/comments/p572xy/rfc_3339_versus_iso_8601/)
    
    In summary, while both formats are based on the ISO 8601 standard, the ISO 20022 DateTime format and RFC 3339 have some differences in their representation of time zones and the precision of fractional seconds, with RFC 3339 being considered more readable by some. The choice between the two formats may depend on the specific requirements of the application or system.

Comment (BorislavSabev) [reply src/GroupHeader.php:86]: @MTijn so I did a lengthy research here. 

Short answer: you are right, kind of :)

Long answer:

ISO20022 two data types for Date and DateTime (ISODate/ISODateTime). Both of those are based on the "XML Schema Part 2: Datatypes Second Edition - W3C
Recommendation 28 October 2004" which is aligned with ISO8601.

Aha! So we should use `\DateTime::ISO8601` or `\DateTime::ISO8601_EXPANDED`. I thought so but nope. 
PHP's `\DateTime::ISO8601` is wrong and is only kept for compatibility reasons. At the same time `\DateTime::ISO8601_EXPANDED` is OK as per ISO8601 but it does not fit our requirements since it uses `An expanded full numeric representation of a year, at least 4 digits, with - for years BCE, and + for years CE. For e.g. -1234 and +10191`. Thus we need to use `DateTime:ATOM` , `DateTime::RFC3339` or `DateTime::W3C` which are all the same. 

As the manual lists `DateTime:ATOM` we should probably stick to that one but in practice any of the 3 are OK. 

I will look into what formats are used throughout this library but you can continue as is in your PR.

File: src/DomBuilder/CustomerDirectDebitTransferDomBuilder.php [hunk 2]
```diff
@@ -271,8 +277,12 @@ public function visitGroupHeader(GroupHeader $groupHeader): void
         if ($groupHeader->getInitiatingPartyId() !== null && in_array($this->painFormat , array('pain.008.001.02','pain.008.003.02'))) {
             $newId = $this->createElement('Id');
             $orgId = $this->createElement('OrgId');
+            $schmeNm = $this->createElement('SchmeNm');
```
Comment (BorislavSabev) [inline src/DomBuilder/CustomerDirectDebitTransferDomBuilder.php:280]: Why are you adding this here? This is an optional element.

**Review by BorislavSabev — CHANGES_REQUESTED** [review]

File: src/DomBuilder/CustomerDirectDebitTransferDomBuilder.php (hunk 2 above)
Comment (MTijn) [reply src/DomBuilder/CustomerDirectDebitTransferDomBuilder.php:280]: I agree according to the specs it is, but our Spanish bank, I am unsure which bank it is will not accept the PAIN file without this information being present.

The moment we add this information all is well according to them.

Comment (rmontagud) [reply src/DomBuilder/CustomerDirectDebitTransferDomBuilder.php:280]: IMHO if it's specific to a single bank entity maybe you should patch the XML file after generating it, if my memory doesn't betray me I had to do that with Triodos because they required an additional element

Comment (MTijn) [reply src/DomBuilder/CustomerDirectDebitTransferDomBuilder.php:280]: We could, but from the issues I read there is a user that has the exact same situation for an Irish bank. We know that the field is marked optional, and in this we we make sure that people can use it when required by their banks. Then why not give the people the option to add it?
This way there would be no need to alter an XML file that is already supposed to be a valid PAIN file.
